web/app/scripts/stepwise/stepwise-directive.js

```diff
@@ -4,154 +4,302 @@
                         .orient('bottom');
+                    return xAxis;
+                }
 
-                    yAxis = d3.svg.axis()
-                        .scale(y)
+                function createYAxis(yScale, data) {
+                    var maxCount = _.max(_.map(data, function(d) {
+                        return d.count;
+                    }));
+                    var yAxis = d3.svg.axis()
+                        .scale(yScale)
                         .orient('left')
-                        .tickSize(1);
-
-                    svg.append('g')
-                      .attr('class', 'yAxis')
-                      .attr('transform', 'translate(' + (isRightToLeft ? width : 0) + ',0)')
-                      .call(yAxis)
-                      .attr('x', -100)
-                      .selectAll('text')
+                        .tickSize(1)
+                        .ticks(maxCount);
+                    return yAxis;
+                }
+
+                function createOuterContainer(svg, containerOffset) {
+                    var outer = svg.append('g')
+                        .attr('class', 'outer')
+                        .attr(
+                            'transform',
+                            'translate(' + containerOffset.x + ',' + containerOffset.y + ')'
+                        );
+                    return outer;
+                }
+
+                function createYAxisContainer(outer, dimensions, offsets, isRightToLeft) {
+                    var yAxisOffset = offsets.yAxis;
+                    var containerOffset = offsets.container;
+                    var yAxisContainer = outer.append('g')
+                        .attr('class', 'yAxis')
+                        .attr('transform', 'translate(' +
+                            (isRightToLeft ? dimensions.width - yAxisOffset.x : yAxisOffset.x) + ', ' +
+                            (yAxisOffset.y - containerOffset.y) + ')');
+                    return yAxisContainer;
+                }
+
+                function addYAxisToContainer(
+                    yAxisContainer, yAxis, dimensions, offsets, isRightToLeft
+                ) {
+                    yAxisContainer.call(yAxis)
+                        .attr('x', -100)
+                        .selectAll('text')
                         .attr('text-anchor', 'right')
-                        .attr('x', -25)
+                        .attr('x', (isRightToLeft ? 5 : -5))
                         .attr('dy', 0);
+                    yAxisContainer
+                        .append('text')
+                        .attr('text-anchor', 'middle')
+                        .attr('transform',
+                            'translate(' +
+                            (isRightToLeft ? offsets.yAxisText.x : -offsets.yAxisText.x) +
+                            ',' + (dimensions.height / 2 - 2 * offsets.yAxis.y) + ')rotate(-90)')
+                        .text($translate.instant('DASHBOARD.STEPWISE_TITLE'));
```

Comment:
This shouldn't appear on the map since the filter period there isn't guaranteed to be 90 days. Currently it does.